yarn-project/circuits.js/src/structs/kernel/public_accumulated_data.ts

```diff
@@ -58,7 +58,7 @@ export class PublicAccumulatedData {
     /**
      * Current public call stack.
      */
-    public publicCallStack: Tuple<CallRequest, typeof MAX_PUBLIC_CALL_STACK_LENGTH_PER_TX>,
+    public readonly publicCallStack: Tuple<CallRequest, typeof MAX_PUBLIC_CALL_STACK_LENGTH_PER_TX>,
```

Comment:
This is great! In the future we might also want to consider using `Readonly*` types (e.g. `ReadonlyArray`, `ReadonlyMap`) to prevent internal modifications